src/test/java/Domain/Users/AssociationRepresentativeTest.java

```diff
@@ -126,12 +125,112 @@ public void removeReferee3ITest() {
         aR = new AssociationRepresentative(aRUser);
         SystemUser newRefereeUser = new SystemUser("refUsername", "refName");
         new Referee(newRefereeUser, "refTraining");
+
+        Referee referee = (Referee) newRefereeUser.getRole(RoleTypes.REFEREE);
+        Season season = new Season(new League("noName"),"2020/21");
+        season.assignReferee(referee);
+        referee.assignToSeason(season);
+
+        assertTrue(season.doesContainsReferee(referee));
         assertTrue(aR.removeReferee(newRefereeUser));
+        assertFalse(season.doesContainsReferee(referee));
+    }
+
+    @Test
+    public void assignRefereeToSeasonUTest() {
+        SystemUser aRUser = new SystemUser("arUsername", "arName");
+        aR = new AssociationRepresentative(aRUser);
+        SystemUser newRefereeUser = new SystemUserStub("refUsername", "refName", 93121);
+        Referee refereeRole = (Referee) newRefereeUser.getRole(RoleTypes.REFEREE);
+        //new RefereeStub(newRefereeUser, "refTraining");
+        Season season = new Season(new League("noName"), "2020/21");
+
+        boolean thrown = false;
+        try {
+            aR.assignRefereeToSeason(season, refereeRole);
+        } catch (Exception e) {
+            thrown = true;
+        }
+        assertFalse(thrown);
+
+        assertTrue(season.doesContainsReferee(refereeRole));
+        assertTrue(refereeRole.getSeasons().contains(season));
+    }
+
+    @Test
+    public void assignRefereeToSeason2UTest() {
+        SystemUser aRUser = new SystemUser("arUsername", "arName");
+        aR = new AssociationRepresentative(aRUser);
+        SystemUser newRefereeUser = new SystemUserStub("refUsername", "refName", 93121);
+        Referee refereeRole = (Referee) newRefereeUser.getRole(RoleTypes.REFEREE);
+        //new RefereeStub(newRefereeUser, "refTraining");
+        Season season = new Season(new League("noName"), "2020/21");
+
+        boolean thrown = false;
+        try {
+            aR.assignRefereeToSeason(season, refereeRole);
+        } catch (Exception e) {
+            thrown = true;
+        }
+        assertFalse(thrown);
+        assertTrue(season.doesContainsReferee(refereeRole));
+        assertTrue(refereeRole.getSeasons().contains(season));
+        //duplicated assign
+        try {
+            aR.assignRefereeToSeason(season, refereeRole);
+        } catch (Exception e) {
+            assertEquals("This referee is already assigned to the chosen season", e.getMessage());
+        }
+    }
```

Comment:
Add "Assert.fail();" at the end of the try, otherwise the test will pass if no exceptions were thrown.

Reply:
צודק הוספתי